docs/architecture.rst

```diff
@@ -161,3 +161,159 @@ the URL makes operating the server more efficient.
 Naturally, we're always looking at ways to improve and reduce the length
 of the URL. This is why it's important to store the entire length of the
 endpoint URL, rather than try and optimize in some manner.
+
+
+.. _protocol:
+
+User Agent Protocol Overview
+============================
+
+.. Remember to update https://github.com/mozilla-services/push-service/docs/design.md
```

Comment:
Nope. Tested it locally and no comments. Likewise the 
    <!-- 
    --> 
in .md docs.

Kinda handy, no?